### ISSUE TITLE
Sqlx Error type

### ISSUE THREAD

**Issue by pkieltyka** [jmoiron/sqlx#130]
It would be nice for sqlx to have its own .Err type that implements the `error` interface. Someone could later cast it and get more details on the error. 

For instance, I'm always getting errors like this:

`sql: Scan error on column index 29: Scan source was not []bytes` .. which tell me the column index, but don't reveal the column name or any other details. It would be nice to at least get the column name with the index. Hmm perhaps this doesn't require a custom error type, maybe it should just include the column name if its available.

**Comment by jmoiron** [comment]
I think that's a `database/sql` error;  I'd have to parse it I think to determine which column it was.  It sounds brittle, but I'm open to suggestions.

**Comment by pkieltyka** [comment]
Yea, that error, and many others come from `database/sql`. The idea is that since sqlx is aware of the index : column name, it could give better hints and improve on the error messages from database/sql, so they are more helpful. One idea would be to have an error mapper, that would proxy error messages from database/sql to callers of sqlx functions. Just a suggestion anyways.

**Comment by kisielk** [comment]
It would probably be helpful if database/sql exposed a `ScanError` type with a `Column` method of field. Currently it just uses `fmt.Errorf`

**Comment by pkieltyka** [comment]
@kisielk I agree, but who knows when something like that would make it into stdlib

**Comment by kisielk** [comment]
Well, if nobody asks for it then never.

**Comment by jszwedko** [comment]
https://groups.google.com/forum/#!forum/golang-nuts would probably be a better venue for a feature request like that than this library.

**Comment by kisielk** [comment]
I agree, hence my suggestion he ask for better error types from database/sql before anything can be done here. Also golang-nuts is not the best place to ask for feature requests either, better to just file a bug in the bug tracker.
